api_tests/src/test_environment.ts

```diff
@@ -0,0 +1,373 @@
+    }
+
+    /**
+     * Start the Lightning Ledger for Bob
+     *
+     * This function assumes that the Bitcoin ledger is initialized.
+     * Once this function returns, the necessary configuration values have been set inside the test environment.
+     */
+    private async startBobLightning() {
+        const config = await this.initLightningLedger("lnd-bob");
+        this.global.lndWallets.bob = await this.initLightningWallet(config);
+    }
+
+    private async initLightningWallet(config: LightningNodeConfig) {
+        return LightningWallet.newInstance(
+            await BitcoinWallet.newInstance(
+                this.global.ledgerConfigs.bitcoin,
+                this.logger
+            ),
+            this.logger,
+            config
+        );
+    }
+
+    private async initLightningLedger(
+        role: string
+    ): Promise<LightningNodeConfig> {
+        const lockDir = await this.getLockDirectory(role);
+        const release = await ledgerLock(lockDir);
+
+        const lnd = await LndInstance.new(
+            await this.global.getDataDir(role),
+            this.logger,
+            await this.global.getDataDir("bitcoind"),
+            path.join(lockDir, "lnd.pid")
+        );
+
+        const config = await this.startLedger(
+            lockDir,
+            lnd,
+            async (lnd) => lnd.config
+        );
+
+        await release();
+
+        return config;
+    }
+
+    private async startLedger<C, S extends LedgerInstance>(
+        lockDir: string,
+        instance: S,
+        makeConfig: (instance: S) => Promise<C>
+    ): Promise<C> {
+        const configFile = path.join(lockDir, "config.json");
+
+        this.logger.info("Checking for config file ", configFile);
+        const configFileExists = await existsAsync(configFile);
+
+        if (configFileExists) {
+            this.logger.info(
+                "Found config file, we'll be using that configuration instead of starting another instance"
+            );
+
+            const config = await readFileAsync(configFile, {
+                encoding: "utf-8",
+            });
+
+            return JSON.parse(config);
+        } else {
+            this.logger.info("No config file found, starting ledger");
+
+            await instance.start();
+
+            const config = await makeConfig(instance);
+
+            await writeFileAsync(configFile, JSON.stringify(config), {
+                encoding: "utf-8",
+            });
+
+            this.logger.info("Config file written to", configFile);
+
+            return config;
+        }
+    }
+
+    private async getLockDirectory(process: string): Promise<string> {
+        const dir = path.join(this.projectRoot, "api_tests", "locks", process);
+
+        await mkdirAsync(dir, {
+            recursive: true,
+        });
+
+        return dir;
+    }
+
+    private static extractLedgersToBeStarted(
+        docblockPragmas: Record<string, string | string[]>
```

Comment:
:question: What is a docblock pragma?

Reply:
The `@foobar` stuff in doc comments above the tests :)